source/dotnet/Library/AdaptiveCards/AdaptiveCard.cs

```diff
@@ -218,5 +218,95 @@ public bool ShouldSerializeJsonSchema()
         {
             return false;
         }
+
+        /// <summary>
+        /// Get resource information for all Images/Media in the whole card
+        /// TODO: Add Media information to the list when Media type is added
+        /// </summary>
+        /// <returns>An array of all card resource information</returns>
+        public RemoteResourceInformation[] GetResourceInformation()
+        {
+            return GetResourceInformationInCard(this).ToArray();
+        }
+
+        private List<RemoteResourceInformation> GetResourceInformationInCard(AdaptiveCard card)
+        {
+            // Initialize the result array
+            List<RemoteResourceInformation> resourceInformationList = new List<RemoteResourceInformation>();
+
+            // Get background image
+            if (!String.IsNullOrEmpty(card.BackgroundImageString))
+            {
+                resourceInformationList.Add(new RemoteResourceInformation(
+                    card.BackgroundImageString,
+                    typeof(AdaptiveImage),
+                    null
+                ));
+            }
+
+            // Get all images in body
+            foreach (AdaptiveElement bodyElement in card.Body)
+            {
+                resourceInformationList.AddRange(GetResourceInformationInElement(bodyElement));
+            }
+
+            // Get all images in ShowCard actions
```

Comment:
actions have icon urls; i checked shared model; its actions elements doesn't do anything; let me check with Becky

Reply:
Right thanks for pointing out